From bbaec8323f2a6930c851d7ffe8461ce6a5280f4f Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Tue, 8 May 2012 13:57:00 +1000
Subject: [PATCH] lib: serial devices may end up with a NULL product_str

If not inputattached, the product_str is NULL and udev doesn't provide a way
to query it. Assume it as 0 instead of crashing.

https://bugzilla.redhat.com/show_bug.cgi?id=819191

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 libwacom/libwacom.c |   16 +++++++++++-----
 1 file changed, 11 insertions(+), 5 deletions(-)

diff --git a/libwacom/libwacom.c b/libwacom/libwacom.c
index 359e945..482baa5 100644
--- a/libwacom/libwacom.c
+++ b/libwacom/libwacom.c
@@ -227,11 +227,17 @@ get_device_info (const char   *path,
 			g_object_unref (old_parent);
 		}
 
-		g_assert (product_str);
-		if (sscanf(product_str, "%d/%x/%x/%d", &garbage, vendor_id, product_id, &garbage) != 4) {
-			libwacom_error_set(error, WERROR_UNKNOWN_MODEL, "Unable to parse model identification");
-			g_object_unref(parent);
-			goto bail;
+		if (product_str) {
+			if (sscanf(product_str, "%d/%x/%x/%d", &garbage, vendor_id, product_id, &garbage) != 4) {
+				libwacom_error_set(error, WERROR_UNKNOWN_MODEL, "Unable to parse model identification");
+				g_object_unref(parent);
+				goto bail;
+			}
+		} else {
+			g_assert(*bus == WBUSTYPE_SERIAL);
+
+			*vendor_id = 0;
+			*product_id = 0;
 		}
 		if (parent)
 			g_object_unref (parent);
-- 
1.7.10.1

